### PR TITLE
Add healthcheck endpoint

### PR DESCRIPTION
**Pull request by jasonpawar-infotrack**
I created an endpoint for health checking the service

Ticket: https://avengerspoc.atlassian.net/browse/AU-23

### CODE REVIEW

File: Unite/Unite/Controllers/ChatController.cs [hunk 1]
```diff
@@ -29,4 +29,11 @@ public ChatController(AssistantService assistantService)
         
         return Ok(response);
     }
+
+    [HttpGet("health")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public IActionResult HealthCheck()
```
Comment (jameskiminfotrack) [inline Unite/Unite/Controllers/ChatController.cs:35]: this is terrible! you no longer family! 😡

File: Unite/Unite/Controllers/ChatController.cs [hunk 2]
```diff
@@ -29,4 +29,11 @@ public ChatController(AssistantService assistantService)
         
         return Ok(response);
     }
+
+    [HttpGet("health")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public IActionResult HealthCheck()
+    {
+        return Ok();
```
Comment (liaminfotrack) [inline Unite/Unite/Controllers/ChatController.cs:37]: what do you return? Don't let me hate you much.

**Review by avengers-poc — COMMENTED** [review]
### Review Comments

The health check endpoint addition looks good overall. Here are a few suggestions for improvement:

1. **Logging**:
   - It would be beneficial to add logging inside the `HealthCheck` action to record when the health check is performed.
     ```csharp
     [HttpGet("health")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     public IActionResult HealthCheck()
     {
         _logger.LogInformation("Health check performed.");
         return Ok();
     }
     ```

2. **Enhance Health Check**:
   - In a real-world application, you might want to check more parameters, like database connectivity or external service health, instead of just returning 200 OK.
     ```csharp
     [HttpGet("health")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status503ServiceUnavailable)]
     public IActionResult HealthCheck()
     {
         bool isHealthy = CheckDatabaseHealth();
         if (isHealthy)
         {
             return Ok(new { Status = "Healthy" });
         }
         else
         {
             return StatusCode(StatusCodes.Status503ServiceUnavailable, new { Status = "Unhealthy" });
         }
     }

     private bool CheckDatabaseHealth()
     {
         // Add logic to check database connection or other dependencies
         return true; // Simplified for illustration
     }
     ```

3. **Unit Tests**:
   - Ensure there are unit tests for this new endpoint to verify its behavior.

**Review by avengers-poc — CHANGES_REQUESTED** [review]
### Review Summary

This PR adds a health check endpoint to the ChatController. While the implementation is mostly correct, there are a few changes and enhancements that should be made to meet the specifications and improve functionality.

### Findings
- **Implemented Path**: The PR implements the endpoint path as `health`.
- **Required Path**: The required endpoint path as per the Jira ticket AU-23 is `checkhealthsystem`.

### Suggested Changes
1. **Endpoint Path Adjustment**: Update the endpoint path to match the requirement from the Jira ticket.
   ```csharp
   [HttpGet("checkhealthsystem")]
   ```
2. **Enhanced Health Information** (optional but recommended): Return more detailed health information to provide a comprehensive health check.

### Recommended Code Adjustment
Adjust the code in `Unite/Unite/Controllers/ChatController.cs` as follows:
```csharp
public class ChatController : ControllerBase
{
    private readonly AssistantService _assistantService;

    public ChatController(AssistantService assistantService)
    {
        _assistantService = assistantService;
    }
    
    // Other methods...
    
    [HttpGet("checkhealthsystem")]
    [ProducesResponseType(StatusCodes.Status200OK)]
    public IActionResult HealthCheck()
    {
        return Ok(new { Status = "Healthy", Timestamp = DateTime.UtcNow });
    }
}
```

This will ensure that the health check endpoint meets the required specifications and provides additional useful information.

File: Unite/Unite/Controllers/ChatController.cs (hunk 2 above)
Comment (avengers-poc) [reply Unite/Unite/Controllers/ChatController.cs:37]: The code currently returns an OK status code with an empty response, as it is a simple health check endpoint. What specific additional information would you recommend including in the response?

File: Unite/Unite/Controllers/ChatController.cs (hunk 1 above)
Comment (avengers-poc) [reply Unite/Unite/Controllers/ChatController.cs:35]: Let us strive to maintain a professional tone during code reviews. Could you kindly provide constructive feedback specifically on the code?

**Pull request closed without merging**